integration-tests/plugins/__tests__/regions/admin/regions.spec.ts

```diff
@@ -0,0 +1,224 @@
+import { useApi } from "../../../../environment-helpers/use-api"
+import { getContainer } from "../../../../environment-helpers/use-container"
+import { initDb, useDb } from "../../../../environment-helpers/use-db"
+import adminSeeder from "../../../../helpers/admin-seeder"
+
+jest.setTimeout(50000)
+
+const env = { MEDUSA_FF_MEDUSA_V2: true }
+const adminHeaders = {
+  headers: { "x-medusa-access-token": "test_token" },
+}
+
+describe("Regions - Admin", () => {
+  let dbConnection
+  let appContainer
+  let shutdownServer
+  let service: IRegionModuleService
+
+  beforeAll(async () => {
+    const cwd = path.resolve(path.join(__dirname, "..", "..", ".."))
+    dbConnection = await initDb({ cwd, env } as any)
+    shutdownServer = await startBootstrapApp({ cwd, env })
+    appContainer = getContainer()
+    service = appContainer.resolve(ModuleRegistrationName.REGION)
+  })
+
+  afterAll(async () => {
+    const db = useDb()
+    await db.shutdown()
+    await shutdownServer()
+  })
+
+  beforeEach(async () => {
+    await adminSeeder(dbConnection)
+
+    // @ts-ignore
+    await service.createDefaultCountriesAndCurrencies()
+  })
+
+  afterEach(async () => {
+    const db = useDb()
+    await db.teardown()
+  })
+
+  it("should create, update, and delete a region", async () => {
+    const api = useApi() as any
+    const created = await api.post(
+      `/admin/regions`,
+      {
+        name: "Test Region",
+        currency_code: "usd",
+      },
+      adminHeaders
+    )
+
+    expect(created.status).toEqual(200)
+    expect(created.data.region).toEqual(
+      expect.objectContaining({
+        id: created.data.region.id,
+        name: "Test Region",
+        currency_code: "usd",
+      })
+    )
+
+    const updated = await api.post(
+      `/admin/regions`,
+      {
+        name: "United States",
+        currency_code: "usd",
+      },
+      adminHeaders
+    )
+
+    expect(updated.status).toEqual(200)
+    expect(updated.data.region).toEqual(
+      expect.objectContaining({
+        id: updated.data.region.id,
+        currency_code: "usd",
+      })
+    )
+
+    const deleted = await api.delete(
+      `/admin/regions/${updated.data.region.id}`,
+      adminHeaders
+    )
+
+    expect(deleted.status).toEqual(200)
+    expect(deleted.data).toEqual({
+      id: updated.data.region.id,
+      object: "region",
+      deleted: true,
+    })
+
+    const deletedRegion = await service.retrieve(updated.data.region.id, {
+      withDeleted: true,
+    })
+
+    // @ts-ignore
+    expect(deletedRegion.deleted_at).toBeTruthy()
+  })
+
+  it("should throw on missing required properties in create", async () => {
+    const api = useApi() as any
+    try {
+      await api.post(`/admin/regions`, {}, adminHeaders)
+    } catch (error) {
+      expect(error.response.status).toEqual(400)
+      expect(error.response.data.message).toEqual(
+        "name must be a string, currency_code must be a string"
+      )
+    }
```

Comment:
I suggest the following here, in general it is better to not have any expectation in the catch because if someone breaks the API and it doesnt throw anymore, then this test will never rich the expectation and will succeed
```suggestion
     const api = useApi() as any
      const err = await api.post(`/admin/regions`, {}, adminHeaders).catch(e => e)
      expect(error.response.status).toEqual(400)
      expect(error.response.data.message).toEqual(
        "name must be a string, currency_code must be a string"
      )
```